public/recorder.js

```diff
@@ -41,9 +56,186 @@ coverImg.appendChild(avatar);
 coverTextInp.value = randomEmoji;
 
 
+function resetAudioGraph() {
+  if (playbackNode) {
+    playbackNode.disconnect();
+    playbackNode.removeEventListener("ended", onEndedHandler);
```

Comment:
is that needed? (out of curiosity)

Reply:
Yes, disconnecting a node from the graph is not enough to stop it, so the "ended" event will still fire. That behavior is specified in the Web Audio API:
> The processing of inputs and the internal operations of an AudioNode take place continuously with respect to AudioContext time, **regardless of whether the node has connected outputs**
> https://webaudio.github.io/web-audio-api/#audionode

I may have missed something but I do not see any way to _destroy_ an `AudioNode`, that is to tell the audio context to forget about it entirely. Weird.

Reply:
let's file an issue in WebAudio? It seems odd there isn't a clean way to destroy a node indeed.

Reply:
I think that the general design in Web Audio is that nodes should have as little "state" or general behavior as possible, because users may create thousands of them in bulk. As such, you can close the context itself, but you cannot close individual nodes. I don't know if it should really be reported as a bug, and the spec is clear that this is expected.

Granted, it is a bit of a nuisance, mostly because there is no `removeAllEventListeners` method, so one needs to keep a pointer to event handlers to remove them so that the node may fall alone in the forest without anyone listening. The question being: is an audio node that falls in a forest without anyone listening still an **audio** node? 🤔

Reply:
Looking at discussions around giving better ergonomics for removing event handlers en masse leads to  https://github.com/whatwg/dom/issues/208 https://github.com/whatwg/dom/issues/412